.github/workflows/docker-build-and-publish.yaml

```diff
@@ -0,0 +1,86 @@
+name: Docker build and publish
```

Comment:
Why is this part of the MR? It already belongs to this MR: https://github.com/bakdata/ci-templates/pull/64

Reply:
it shouldnt be part of this PR